packages/contract/lib/utils/index.js

```diff
@@ -80,6 +80,60 @@ const Utils = {
       .filter(log => log != null);
   },
 
+  consoleLog(receiptLogs) {
+    const logEvents = receiptLogs.filter(x => x.event === "_TruffleConsoleLog");
+    logEvents.forEach(event => {
+      const result = event.args;
+
+      // A mapping to show how to output into our console
+      const outputMap = {
+        num: result[0].toString(),
+        boolean: result[0],
+        str: result[0],
+        b32: result[0],
+        addr: result[0]
+      };
+
+      // A list of all "supported" types: ["num", "boolean", "str", ...]
+      const types = Object.keys(outputMap);
+
+      // Find out what type our result is, and log it out if it matches
+      types.forEach(type => {
+        if (result.hasOwnProperty(type)) {
+          console.log(outputMap[type]);
+        }
+      });
+    });
+
+    const namedLogEvents = receiptLogs.filter(
+      x => x.event === "_TruffleConsoleLogNamed"
+    );
+    namedLogEvents.forEach(event => {
+      const result = event.args;
+
+      // A mapping to show how to output into our console
```

Comment:
Hmm, I can't really think of an elegant way of doing this other than creating an awkward function because both of these are in closures. In addition, you'd have to pass the index as well which would be ugly. Any ideas?